api/src/services/lottery.service.ts

```diff
@@ -1113,31 +1132,71 @@ export class LotteryService {
       throw new ForbiddenException();
     }
 
-    const applicationUserId = await this.prisma.applications.findFirstOrThrow({
+    if (!user.userRoles.isAdmin) {
+      const applicationUserId = await this.prisma.applications.findFirstOrThrow(
+        {
+          select: {
+            userId: true,
+          },
+          where: {
+            id: applicationId,
+          },
+        },
+      );
+
+      await this.permissionService.canOrThrow(
+        user,
+        'application',
+        permissionActions.read,
+        {
+          userId: applicationUserId.userId,
+        },
+      );
+    }
+
+    const results = await this.prisma.applicationLotteryPositions.findMany({
       select: {
-        userId: true,
+        ordinal: true,
+        multiselectQuestionId: true,
       },
       where: {
-        id: applicationId,
+        applicationId,
       },
     });
 
-    await this.permissionService.canOrThrow(
-      user,
-      'application',
-      permissionActions.read,
-      {
-        userId: applicationUserId.userId,
-      },
-    );
+    return results;
+  }
 
-    const results = await this.prisma.applicationLotteryPositions.findMany({
+  /*
+   * @param id - listing id
+   * @returns an array of totals
+   */
+  public async lotteryTotals(
+    listingId: string,
+    user: User,
+  ): Promise<ApplicationLotteryTotal[]> {
+    if (!user) {
+      throw new ForbiddenException();
+    }
+
+    if (!user.userRoles.isAdmin) {
+      await this.prisma.applications.findFirstOrThrow({
```

Comment:
prisma errors are very hard to parse, we shouldn't rely on their `orThrow()` because troubleshooting them in production will be pretty hard. 

instead I recommend doing what we do in other services where we attempt to grab the record and then do a null check and throw an error we can understand if its missing

Reply:
Gotcha! I was copying patterns from elsewhere, particularly `lotteryStatus` - we have ~30 of instances of `findOrThrow`, should we make a ticket to remove them all?

Reply:
If I'm not mistaken all instances of `findOrThrow` are custom functions to encapsulate the logic

if you peep `ami-chart.service.ts ~line 137` you'll notice its a custom function that does a findUnique() and then checks to see

so those should be okay. Its more an issue with the inbuilt orThrow functions in prisma

Reply:
Will take a look! If you have space let's peek at this in a huddle